docs/developing.md

```diff
@@ -9,46 +9,94 @@ $ cd $(go env GOPATH)/src/github.com/deislabs/duffle
 ```
 
 ### Prerequisites
+
 You need:
-* A working Go 1.11.4 (or later) environment
+
 * make
+* Docker
 
-Before you start working with the code, issue:
-```console
-$ make bootstrap
-```
+## Containerized Development Environment
+
+To ensure a consistent development environment for all contributors, Duffle
+relies heavily on Docker containers as sandboxes for all development activities
+including dependency resolution and executing tests.
+
+`make` targets seamlessly handle the container orchestration.
+
+If, for whatever reason, you must opt-out of executing development tasks within
+containers, set the `SKIP_DOCKER` environment variable to `true`, but be aware
+that by doing so, the success or failure of development-related tasks, tests,
+etc. will be dependent on the state of your system, with no guarantee of the
+same results in CI.
+
+## Developing on Windows
+
+All development-related tasks should "just work" on Linux and Mac OS systems.
+When developing on Windows, the maintainers strongly recommend utilizing the
+Windows Subsystem for Linux.
+
+[This blog post](https://nickjanetakis.com/blog/setting-up-docker-for-windows-and-wsl-to-work-flawlessly)
+provides excellent guidance on making the Windows Subsystem for Linux work
+seamlessly with Docker Desktop (Docker for Windows).
 
 ## Building
 
-To build `bin/duffle`, issue:
+To build everything (binaries for Linux, Mac, and Windows on amd64 architecture)
+as well as a linux/amd64 Docker image containing the corresponding binary:
+
 ```console
 $ make build
 ```
 
-If you want to install `duffle`, issue:
+To build binaries for Linux, Mac, and Windows, but no Docker image:
+
+```console
+$ make build-all-bins
+```
+
+To build for one specific OS / architecture:
+
 ```console
-$ sudo make install
+$ OS=<desired OS> ARCH=<desired architecture> make build-bin
```

Comment:
This is part of my local development cycle, so would it be possible to add a simple alias such as `make duffle` which would produce `bin/duffle`?

Reply:
I struggled with what to do about that target. It doesn't work on Windows...

Reply:
Ok... more thought into this...

Rather than include a target that won't work across all OSes, can I suggest contributors who previously took advantage of this target may actually get _more_ mileage from a symlink from `/usr/local/bin/duffle` (or elsewhere on the path) to `$GOPATH/src/github.com/deislabs/duffle/bin/duffle-<OS>-amd64`?

With this approach, there is no separate step required to install after rebuilding.

Reply:
Yes, makes sense.